ir/instruction.ts

```diff
@@ -0,0 +1,162 @@
+export interface LiteralValue {
+  num?: number;
+  id?: string;
+  coord?: [number, number];
+}
+
+export function isLiteralValue(x: unknown): x is LiteralValue {
+  return (
+    typeof x === "object" &&
+    x != null &&
+    (x as LiteralValue).num !== undefined ||
+    (x as LiteralValue).id !== undefined ||
+    (x as LiteralValue).coord !== undefined
+  );
+}
+
+interface Label {
+  label: string;
+}
+
+export function parseLabel(x: string): Label {
+  if (x.startsWith(":")) {
+    return { label: x.substring(1) };
+  } else if (x.endsWith(":")) {
```

Comment:
Where are we using labels ending with ':' ?

Reply:
I meant this to work for a label for a line in assembly and a label ref as an argument. But i never ended up using this function